packages/devtools_app/lib/src/common_widgets.dart

```diff
@@ -537,6 +537,62 @@ class ExportButton extends StatelessWidget {
   }
 }
 
+class FilterButton extends StatelessWidget {
+  const FilterButton({
+    Key key,
+    @required this.onPressed,
+    @required this.isFilterActive,
+  }) : super(key: key);
+
+  final VoidCallback onPressed;
+
+  final bool isFilterActive;
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    return RoundedOutlinedBorder(
+      child: SizedBox(
+        height: defaultButtonHeight,
+        child: Tooltip(
+          message: 'Filter',
+          child: FlatButton(
+            key: key,
+            onPressed: onPressed,
+            color: isFilterActive
+                ? colorScheme.toggleButtonBackgroundColor
+                : Colors.transparent,
+            child: createIcon(
+              Icons.filter_list,
+              color: isFilterActive
+                  ? colorScheme.toggleButtonForegroundColor
+                  : null,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+Widget clearTextFieldButton(VoidCallback onPressed) {
+  return textFieldSuffixButton(Icons.clear, onPressed);
+}
+
+Widget textFieldSuffixButton(IconData icon, VoidCallback onPressed) {
+  return Container(
+    padding: const EdgeInsets.symmetric(horizontal: densePadding),
+    width: 24.0,
+    child: IconButton(
+      padding: const EdgeInsets.all(0.0),
```

Comment:
nit: would placing the padding in the icon button look different than in the container?

Reply:
the explicit 0 padding is to override the default padding of 4.0 in the IconButton, otherwise the size constraint of 24.0 is not respected.